src/main/java/software/amazon/awssdk/crt/http/Http2Request.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.awssdk.crt.http;
+
+import software.amazon.awssdk.crt.http.HttpClientConnection.ProtocolVersion;
+
+/**
+ * Represents a single Client Request to be sent on a HTTP connection
+ */
+public class Http2Request extends HttpRequestBase {
```

Comment:
When calling `makeRequest` with a normal `HttpRequest` like this:

```
HttpRequest request = new HttpRequest("GET", "/");
http2Connection.makeRequest(request, ...);
```

Does CRT do all of the conversions for me into the HTTP/2 request; for example setting the `:method` and `:path` headers?

Reply:
I just did some testing against Kinesis and it doesn't look like it :(. Can this be supported? Again, the goal I think is that `HttpRequest` can be constructed and executed without the user having to worry whether they're working with H2 or HTTP/1.1 under the hood.

Reply:
We do support converting a HTTP/1 request to HTTP/2, as long as you created a HTTP/2 connection. I guess you are not getting a HTTP/2 connection.